addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -584,15 +612,42 @@ var FileWidget = SearchableMediaWidget.extend({
                             'res_model': self.options.res_model,
                             'width': 0,
                             'quality': 0,
+                        }}, {
+                            xhr: function () {
+                                var xhr = $.ajaxSettings.xhr();
+                                xhr.upload.onprogress = function (ev) {
+                                    var prcComplete = ev.loaded / ev.total * 100;
+                                    $progressBar.find('.progress-bar').css({
+                                        width: parseInt(prcComplete) + '%',
+                                    }).text(prcComplete.toFixed(2) + '%');
+                                };
+                                xhr.upload.onload = function () {
+                                    // Don't show yet success as backend code only starts now
+                                    $progressBar.find('.progress-bar').css({width: '100%'}).text('100%');
+                                };
+                                return xhr;
+                            },
```

Comment:
Cool feature but should work with unsplash download too :)

Reply:
Done, I kept the changes required for unsplash integration in a separate commit, let me know if you want it to be merged with the commit introducing the feature for regular image upload.
Note: it was maybe not obvious but the feature was also done for files/documents upload